staging/src/k8s.io/dynamic-resource-allocation/controller/controller_test.go

```diff
@@ -340,6 +347,10 @@ func TestController(t *testing.T) {
 			) {
 				t.Fatal("could not sync caches")
 			}
+			cc := ctrl.(*controller)
+			// We need to mock the event recorder to test the controller's event.
+			fakeRecorder := record.NewFakeRecorder(100)
+			cc.eventRecorder = fakeRecorder
```

Comment:
Ah, so that explains the `client-go/tools/record/event.go` failure, right?

Good find, thanks!

Reply:
yes